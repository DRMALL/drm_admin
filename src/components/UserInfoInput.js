

import React from 'react'
import Row from 'antd/lib/row'
import Col from 'antd/lib/col'
import Input from 'antd/lib/input'
import store from '../commons/store'
import { userInput } from '../commons/userCommon'
import getInputValue from '../actions/getInputValue'

const UserInfoInput = () => (
    <div>
        {
          userInput.map((item, index) =>
          <Row key={ index } className='new-user-item'  >
            <Col span={3} className='new-user-item-left' >{item.title}</Col>
            <Col span={7} ><Input data-id={ item.id } data-path={ item.path }
                        onChange={ getInputValue }
                        value={ changeValue(item.id) }
                        placeholder={ `输入${item.title}` } /></Col>
          </Row> )
        }
    </div>
  )

export default UserInfoInput

function changeValue(str){
  const { name, password, email, phone, company_name, address } = store.getState().user
  switch(str){
    case 'name':
      return name
    case 'password':
      return password
    case 'email':
      return email
    case 'phone':
      return phone
    case 'company_name':
      return company_name
    case 'address':
      return address
    default:
      return ''
  }
}
