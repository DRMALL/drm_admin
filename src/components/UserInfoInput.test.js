

import React from 'react'
import renderer from 'react-test-renderer'
import UserInfoInput from './UserInfoInput'

it('UserInfoInput snapshot test', () => {
  const tree = renderer.create(<UserInfoInput />)
  expect(tree).toMatchSnapshot()
})
