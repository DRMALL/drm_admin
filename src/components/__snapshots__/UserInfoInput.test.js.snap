// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoInput snapshot test 1`] = `
<div>
  <div
    className="ant-row new-user-item"
    style={undefined}
  >
    <div
      className="ant-col-3 new-user-item-left"
    >
      用户名
    </div>
    <div
      className="ant-col-7"
    >
      <input
        className="ant-input"
        data-id="name"
        data-path="USER"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="输入用户名"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ant-row new-user-item"
    style={undefined}
  >
    <div
      className="ant-col-3 new-user-item-left"
    >
      密码
    </div>
    <div
      className="ant-col-7"
    >
      <input
        className="ant-input"
        data-id="password"
        data-path="USER"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="输入密码"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ant-row new-user-item"
    style={undefined}
  >
    <div
      className="ant-col-3 new-user-item-left"
    >
      用户邮箱
    </div>
    <div
      className="ant-col-7"
    >
      <input
        className="ant-input"
        data-id="email"
        data-path="USER"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="输入用户邮箱"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ant-row new-user-item"
    style={undefined}
  >
    <div
      className="ant-col-3 new-user-item-left"
    >
      联系电话
    </div>
    <div
      className="ant-col-7"
    >
      <input
        className="ant-input"
        data-id="phone"
        data-path="USER"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="输入联系电话"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ant-row new-user-item"
    style={undefined}
  >
    <div
      className="ant-col-3 new-user-item-left"
    >
      公司名称
    </div>
    <div
      className="ant-col-7"
    >
      <input
        className="ant-input"
        data-id="company_name"
        data-path="USER"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="输入公司名称"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ant-row new-user-item"
    style={undefined}
  >
    <div
      className="ant-col-3 new-user-item-left"
    >
      通讯地址
    </div>
    <div
      className="ant-col-7"
    >
      <input
        className="ant-input"
        data-id="address"
        data-path="USER"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="输入通讯地址"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
